src/wasm-binary-reader-interpreter.c

```diff
@@ -527,15 +529,17 @@ static WasmResult begin_function_body(uint32_t index, void* user_data) {
 
 static WasmResult end_function_body(uint32_t index, void* user_data) {
   Context* ctx = user_data;
-
-  /* discard everything except the last expr */
-  uint32_t discard_max = ctx->expr_stack.size - 1;
+  uint32_t discard_max = 0;
 
   WasmInterpreterFuncSignature* sig =
       get_func_signature(ctx, ctx->current_func);
   if (sig->result_type == WASM_TYPE_VOID) {
     /* discard last expr too */
```

Comment:
This comment makes less sense with the other comment removed. Can you either add new comments or remove this one?